lib/builder.js

```diff
@@ -235,9 +239,9 @@ module.exports = class Builder extends EventEmitter {
     } else {
       // nodeType === 'source'
       if (nodeInfo.watched) {
-        this.watchedPaths.push(nodeInfo.sourceDirectory);
+        this.watchedPaths.push({ root: nodeWrapper, path: nodeInfo.sourceDirectory });
```

Comment:
I think we should store sourceNodes as a separate property so as not to break the public interface of watchedPaths.